designs/DotNetSignCommands.md

```diff
@@ -0,0 +1,140 @@
+
+EXAMPLES:
+
+    dotnet nuget sign MyPackage.nupkg --timestamper https://foo.bar
+
+    dotnet nuget sign ../MyPackage.nupkg --timestamper https://foo.bar --output-directory ../signed
+```
+
+Notable deviations from `nuget.exe` version:
+
+- No `-ConfigFile` equivalent. `dotnet.exe` doesn't seem to do this at all.
+- No `-ForceEnglishOutput` equivalent. `dotnet.exe` doesn't seem to do this at all.
+- Use MSBuild verbosity levels for `--verbosity` instead of NuGet levels.
+- Use `--interactive` instead of `--non-interactive`, because that's how `dotnet.exe` does it.
+
+### `dotnet nuget verify`
+
+```
+Verifies a signed NuGet package at <package-path>.
+
+USAGE:
+    dotnet nuget verify [OPTIONS] <package-path>...*
+
+OPTIONS:
+        --all-verifications*                    Specifies that all verifications possible should be performed on the package(s). By default, only signatures are verified.
+        --certificate-fingerprint <fingerprint> Verify that the signer certificate matches with one of the specified SHA256 fingerprints. This option may be supplied multiple times to provide multiple fingerprints.
+    -h, --help                                  Prints help information.
+        --interactive*                          Allows command to stop and wait for user input or action.
+```
+
+Notable deviations from `nuget.exe` version:
+
+- `-Signatures` has been removed and is now default. Use `--all-verifications` to perform `-All`.
+- Multiple position arguments can be passed in to verify multiple packages, instead of only supporting globbing.
+- No `-ConfigFile` equivalent. `dotnet.exe` doesn't seem to do this at all.
+- No `-ForceEnglishOutput` equivalent. `dotnet.exe` doesn't seem to do this at all.
+- Use MSBuild verbosity levels for `--verbosity` instead of NuGet levels.
+- Use `--interactive` instead of `--non-interactive`, because that's how `dotnet.exe` does it.
+
+## Future Work
+
+For now, we're not doing `trusted-signer`, and specifying all the behavior
+probably deserves its own spec because of the complexity of the command and
+the ergonomic expectations of `dotnet.exe`. Below is an initial stab at a
+design that should NOT be considered normative, but establishes some
+direction. Most notably, it splits the very complicated `add`, which has
+multiple gestures, into multiple, more specific subcommands with their own
+options and behavior.
+
+### `dotnet nuget trusted-signer`
+
+```
+Provides the ability to manage the list of trusted signers.
+
+USAGE:
+    dotnet nuget trusted-signer [OPTIONS] [COMMAND]
+
+COMMANDS:
+    list (default)
+    add-repository <name> <repository> [<package>...]
+    add-author <name> <author> [<package>...]
+    add-service-index <name> <index>
+    add-certificate <name> <fingerprint>
+    remove <name>
+    sync <name>
+```
+
+## Open Questions
+
+Is there any precedent at all in `dotnet.exe` for a `--config-file` option? Anywhere?
```

Comment:
`dotnet nuget add source` and `dotnet nuget add client-cert` (although it uses `--configfile`, which I think seems incorrect, but I don't know if we can change now).

Also, not `--config-file` specifc, but `dotnet sln [path\to\file.lsn] add path\to\project.csproj` is semantically the same, as is `dotnet add [path\to\project.csproj] package package_id`. Just as `dotnet add` and `dotnet sln add`  modify files and allow dotnet to either automatically detect the file, or allow you to specify the target, `dotnet nuget trusted-signer` is all about modifying a config file.